probe-rs/src/architecture/arm/traits.rs

```diff
@@ -33,36 +86,45 @@ bitfield::bitfield! {
     pub swclk_tck, set_swclk_tck: 0;
 }
 
-/// Debug port address.
-#[derive(Debug, Eq, PartialEq, PartialOrd, Ord, Clone, Copy, Hash, Default)]
-pub enum DpAddress {
-    /// Access the single DP on the bus, assuming there is only one.
-    /// Will cause corruption if multiple are present.
-    #[default]
-    Default,
-    /// Select a particular DP on a SWDv2 multidrop bus. The contained `u32` is
-    /// the `TARGETSEL` value to select it.
-    Multidrop(u32),
-}
-
 /// Access port v2 address
 #[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone, Hash)]
-pub enum ApV2Address {
-    /// Last node of an APv2 address
-    Leaf(u32),
-    /// Non-terminal node of an APv2 address
-    Node(u32, Box<ApV2Address>),
+pub struct ApV2Address(Vec<u64>);
+
+impl Default for ApV2Address {
+    fn default() -> Self {
+        Self::new()
+    }
 }
 
-impl std::fmt::Display for ApV2Address {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        match self {
-            ApV2Address::Leaf(v) => write!(f, "{}", v),
-            ApV2Address::Node(v, r) => write!(f, "{}.{}", v, r),
-        }
+impl ApV2Address {
+    /// Create a new ApV2 Address chain
+    pub fn new() -> Self {
+        Self(Vec::new())
```

Comment:
What is the meaning of an empty address? What AP is addressed by it?

Reply:
From what I understand of the changes, the empty address is used to refer to the root AP of the DAP register space.

Reply:
Updated to make this function `root()` for clarity in 27d893af4479d1ce04d8f33d7c40987e8e7e49f4